packages/vitest/src/utils/coverage.ts

```diff
@@ -537,20 +537,32 @@ export class BaseCoverageProvider<Options extends ResolvedCoverageOptions<'istan
     const servers = [
       ...ctx.projects.map(project => ({
         root: project.config.root,
+        browser: project.browser?.vite,
         vitenode: project.vitenode,
       })),
       // Check core last as it will match all files anyway
-      { root: ctx.config.root, vitenode: ctx.vitenode },
+      { root: ctx.config.root, vitenode: ctx.vitenode, browser: ctx.getRootProject().browser?.vite },
     ]
 
     return async function transformFile(filename: string): Promise<TransformResult | null | undefined> {
       let lastError
 
-      for (const { root, vitenode } of servers) {
-        if (!filename.startsWith(root)) {
+      for (const { root, vitenode, browser } of servers) {
+        // On Windows root doesn't start with "/" while filenames do
+        if (!filename.startsWith(root) && !filename.startsWith(`/${root}`)) {
```

Comment:
On Windows the `root` is here `D:/a/vitest/vitest/test/coverage-test` while filename is `/D:/a/vitest/vitest/test/coverage-test/fixtures/src/untested-file.ts`.